gnucash/gnome/gnc-plugin-page-account-tree.c

```diff
@@ -1460,11 +1460,11 @@ gnc_plugin_page_account_tree_cmd_delete_account (GtkAction *action, GncPluginPag
             g_free(acct_name);
             return;
         }
-        if (trans_mas && gtk_widget_is_sensitive(trans_mas))
+        if (GTK_IS_WIDGET(trans_mas) && gtk_widget_is_sensitive(trans_mas))
```

Comment:
The first line of gtk_widget_is_sensitive() is
```c
g_return_if_fail(GTK_IS_WIDGET(widget), FALSE);
```
If widget isn't NULL (G_IS_OBJECT checks for NULL and returns false) but is an invalid address then all you've done is move the crash from gtk_widget_is_sensitive() to here.

Where do you think that the widgets are getting destroyed?

Reply:
I think I may of confused my self but there is a gtk_widget_destroy on "subaccount_trans" GtkBox which encompasses the hbox that the Gas widget is packed and so assumed that was destroyed also.

Reply:
Well, the logic in that function isn't great. If the child accounts have no splits then indeed subaccount_trans gets destroyed, after which sa_trans_mas is created and packed into the destroyed box. trans_mas and sa_mas, on the other hand, are part of other components that aren't destroyed. The invalid pointer that causes the crash in the bug is indeed sa_trans_mas, so perhaps the problem is its unconditional creation; even if it doesn't crash it still leaks. So I think the fix should be to rearrange the subaccount_trans logic: If the subaccount has transactions, then create sa_trans_mas and do the read-only check, else delete subaccount_trans.

Incidentally, GTK_IS_WIDGET(sa_trans_mas) should return TRUE; it's freshly allocated, after all. gtk_widget_is_sensitive is probably crashing as it climbs the chain of parents; parent->parent is undefined after the parent's destruction.

Reply:
I wondered about changing the order so will have another stab at this.

Reply:
This dialog is strange, the vbox subaccount_trans which contains Gas widget sa_trans_mas is never sensitive / greyed out so there is no point in it being there as far as I can see. I thought it was a mistake on unstable but it is the same on maint.
Should this section "Sub-Account Transactions" get removed from the dialog and associated code or am I missing something ?

Reply:
ISTM it's together with the messed-up logic in the if statements: The subaccount_trans box should be made sensitive if there are subaccount transactions.

But IIRC it's a HIG failure to delete GUI elements anyway, they should just be made insensitive. So maybe the gtk_wdget_destroy calls should be removed.

Reply:
What I was getting at, if the account you are deleting has sub accounts and you specify to move them why would you move any sub account transactions some where else.

Reply:
The dialog comprises of a vbox containing three vbox's , 'transactions', 'subaccounts' and 'subaccount-trans'. What I think should be displayed is if there are transactions in the deleted account 'transactions' is left, if there are sub accounts then you may have 'transactions' and 'subaccounts' or you may have 'transctions' and 'subaccount-trans'.

Reply:
OK. Will it be ready by tomorrow morning so that I can release 2.7.8 tomorrow?